pkg/kubectl/cmd/scale.go

```diff
@@ -122,43 +121,35 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 		return err
 	}
 
-	infos, err := r.Infos()
-	if err != nil {
-		return err
-	}
-	info := infos[0]
-	mapping := info.ResourceMapping()
-	scaler, err := f.Scaler(mapping)
-	if err != nil {
-		return err
-	}
+	counter := 0
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
 
-	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
-	if len(resourceVersion) != 0 && len(infos) > 1 {
-		return fmt.Errorf("cannot use --resource-version with multiple resources")
```

Comment:
Ah, true. Ok thanks

On Thursday, April 28, 2016, David Eads <notifications@github.com> wrote:

> In pkg/kubectl/cmd/scale.go
> <https://github.com/kubernetes/kubernetes/pull/23673#discussion_r61475083>
> :
>
> >
> > -	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
> > -	if len(resourceVersion) != 0 && len(infos) > 1 {
> > -		return fmt.Errorf("cannot use --resource-version with multiple resources")
>
> Do you still think these checks are still necessary given that the usage
> of infos is being superseded by using Visit()?
>
> If someone specifies --resourceversion for multiple objects, you
> guaranteed that it cannot succeed on more than one of them.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23673/files/c04de61a6caf10487672f05b72a1c81b5a83061d#r61475083>
>


-- 
-Mike Metral